pkg/pipeline/resource_pipeline_node_pool.go

```diff
@@ -85,24 +87,24 @@ func pipelineNodePoolResource() *schema.Resource {
 	}
 
 	var unpackNodePool = func(data *schema.ResourceData) (NodePool, error) {
-		d := &ResourceData{data}
+		d := &util.ResourceData{data}
 
 		nodePool := NodePool{
-			ProjectId:              d.getInt("project_id"),
-			Name:                   d.getString("name"),
-			NumberOfNodes:          d.getInt("number_of_nodes"),
-			IsOnDemand:             d.getBool("is_on_demand"),
-			Architecture:           d.getString("architecture"),
-			OperatingSystem:        d.getString("operating_system"),
-			NodeIdleIntervalInMins: d.getInt("node_idle_interval_in_mins"),
-			Environments:           d.getList("environments"),
+			ProjectId:              d.GetInt("project_id", false),
+			Name:                   d.GetString("name", false),
+			NumberOfNodes:          d.GetInt("number_of_nodes", false),
+			IsOnDemand:             d.GetBool("is_on_demand", false),
+			Architecture:           d.GetString("architecture", false),
```

Comment:
Is there strict validation on this?

Reply:
I'm not planning to update the code in this PR.